miner/bidder.go

```diff
@@ -0,0 +1,265 @@
+package miner
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	dialer = &net.Dialer{
+		Timeout:   time.Second,
+		KeepAlive: 60 * time.Second,
+	}
+
+	transport = &http.Transport{
+		DialContext:         dialer.DialContext,
+		MaxIdleConnsPerHost: 50,
+		MaxConnsPerHost:     50,
+		IdleConnTimeout:     90 * time.Second,
+		TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
+	}
+
+	client = &http.Client{
+		Timeout:   5 * time.Second,
+		Transport: transport,
+	}
+)
+
+type ValidatorConfig struct {
+	Address common.Address
+	URL     string
+}
+
+type BidderConfig struct {
+	Enable     bool
+	Validators []ValidatorConfig
+	SecretKey  string
+}
+
+type Bidder struct {
+	config *BidderConfig
+	engine consensus.Engine
+	chain  *core.BlockChain
+
+	validators map[common.Address]*ethclient.Client // validator address -> ethclient.Client
+
+	bestWorksMu sync.RWMutex
+	bestWorks   map[int64]*environment
+
+	builderSecretKey *ecdsa.PrivateKey
```

Comment:
`PrivateKey` is not suggested, try same pattern like parlia?

Reply:
Manage builder account with geth?

Reply:
Done